lib/parsers/mp3_parser.rb

```diff
@@ -47,10 +47,6 @@ def to_h
         h[k] = value if value
       end
     end
-
-    def as_json(*)
-      to_h
-    end
```

Comment:
It looks like `mp3_parser` had a specialised `as_json` implementation which is now removed?

Perhaps it should instead be something like (pseudo code):
```ruby
def as_json(stringify_keys: false)
  h = to_h
  h = FormatParser::HashUtils.deep_transform_keys(h, &:to_s) if stringify_keys
  h
end
```
Although it shows that the `root` argument was already unsupported by `mp3_parser`...

Reply:
This method that I removed was defined in `TagWrapper` which is used only in this piece of code:

```ruby
# lib/parsers/mp3_parser.rb:279
def blend_id3_tags_into_hash(*tags)
  tags.each_with_object({}) do |tag, h|
    h.merge!(TagWrapper.new(tag).to_h)
  end
end
```

Given the fact it returns a `Hash` (and not a `TagWrapper` instance), I'm almost sure that `as_json` is not being used anywhere.

But there is a risk someone is using the class `TagWrapper` directly, but I don't think so, because there is no info about it on README.

Before deleting this method, I ran all the tests to make sure nothing reached there (but I agree this is not a guarantee)

I chose to remove the method instead of using `deep_transform_keys` as you suggested, but I agree it would be a safer approach.

wdyt?

Reply:
I see, thanks for the explanation! I'll defer to someone more familiar with the codebase to comment on that.

Reply:
Maybe double check with Julik, see if he has more to add on this.

Reply:
The idea with this was to ensure that no id3tag gem structures bleed into the FormatParser result, and thus can be serialized to JSON in a simpler/more transparent way. So I think this should work fine.